### DISCUSSION TITLE
访问系统时，经常超时？

### DISCUSSION THREAD

**Discussion by zuihou** [zuihou/zuihou-admin-cloud#254] [01.常见问题]
微服务项目中，不同服务之间的调用是通过 http 调用的方式来请求的，如： 请求经过 网关转发给其他服务， 服务使用Feign进行接口调用， 所以如果你使用的nacos中，同时注册了多个 **同名服务**，请求时就会出现超时的情况。

1. 检查 nacos - 服务中心 - 实例数
    ![image](https://user-images.githubusercontent.com/8500410/227817861-1e6e958d-7b05-4f7b-a997-c4372d28408a.png)
2. 若同一个服务中，实例数 超过1个，需要保证所有的实例 通过 下图中的IP能互相访问、
    ![image](https://user-images.githubusercontent.com/8500410/227817984-d385cf2f-5442-4451-a9fe-9c129dd96fcf.png)

## 异常情况1：
    若注册的实例中，实例1 部署在 云服务器，实例2 在你本地电脑启动， 访问时大概率会1次超时，1次正常。
## 原因：
    阿里云服务器部署的服务，注册到nacos时，使用的是阿里云服务器的内网IP，与你本机电脑所处的网络是不互通的，所以出出现1次请求正常，1次请求超时。

## 异常情况2：
   若注册的实例中，实例1 在局域网内其他开发者电脑，实例2 在你本地电脑， 访问时大概率会时而超时，时而正常。
## 原因：
    多个开发者同时注册到1个nacos时， 其他开发者可能在debug等等情况，导致无规律的出现超时情况。

## 解决方案
1. 新建nacos的命名空间区分开发、测试、uat环境， 阿里云上的服务单独使用一个 命名空间、不同的开发者使用自己独立的命名空间
2. 不同的开发者自己搭建本地nacos，不用与他人共用nacos
3. 在nacos控制台，将同一个服务的其他实例 **下线** （其他开发者可能会打你）

**Comment by hellon** [comment]
nacos现在是在我本地搭建的仍然还是会出现超时的情况。
![image](https://user-images.githubusercontent.com/15033782/227852731-8a08ec26-ea29-4eb5-87b7-c3bcc191bff3.png)

超时
![image](https://user-images.githubusercontent.com/15033782/227852767-d5b528a5-145b-40db-a4b5-962c4a92b475.png)

**Reply by zuihou** [reply]
这个超时是前端超时， 请排查后端 gateway服务和system服务的是否有报错日志，  若没有， 请在gateway 的 TokenContextFilter 和 system 服务报错的方法 中打印日志， 看是否正常访问。

并用postman调用接口，看是否正常。

**Comment by hellon** [comment]
出现的情况不止一次，还有没有其他的方式
